src/NuGet.Core/NuGet.ProjectModel/CacheFileFormat.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+
+namespace NuGet.ProjectModel
+{
+    public class CacheFileFormat
+    {
+        private const string DGSpecHashProperty = "dgSpecHash";
+
+        public static CacheFile Read(string filePath)
+        {
+            return Read(filePath, NullLogger.Instance);
+        }
+
+        public static CacheFile Read(string filePath, ILogger log)
+        {
+            using (var stream = File.OpenRead(filePath))
+            {
+                return Read(stream, log, filePath);
+            }
+        }
+
+        private static CacheFile Read(Stream stream, ILogger log, string path)
+        {
+            using (var textReader = new StreamReader(stream))
+            {
+                return Read(textReader, log, path);
+            }
+        }
+
+        private static CacheFile Read(TextReader reader, ILogger log, string path)
+        {
+            try
+            {
+                using (var jsonReader = new JsonTextReader(reader))
+                {
+                    while (jsonReader.TokenType != JsonToken.StartObject)
+                    {
+                        if (!jsonReader.Read())
+                        {
+                            throw new InvalidDataException();
+                        }
+                    }
+                    var token = JToken.Load(jsonReader);
+                    var cacheFile = ReadCacheFile(token as JObject);
+                    return cacheFile;
+                }
+            }
+            catch (Exception ex)
+            {
+                log.LogError(string.Format(CultureInfo.CurrentCulture,
+                    "error reading a cache file {1} : {2}",
```

Comment:
Logging this as an error may be confusing since it doesn't look like this will actually fail the restore. I would suggest making this a warning or diagnostic output. Worst case we just run the restore.